airflow/secrets/base_secrets.py

```diff
@@ -40,10 +47,46 @@ def build_path(path_prefix: str, secret_id: str, sep: str = "/") -> str:
         """
         return f"{path_prefix}{sep}{secret_id}"
 
+    def get_conn_value(self, conn_id: str) -> Optional[str]:
+        """
+        Retrieve from Secrets Backend a string value representing the Connection object.
+
+        If the client your secrets backend uses already returns a python dict, you should override
+        ``get_connection`` instead.
+
+        :param conn_id: connection id
+        """
+        raise NotImplementedError
+
+    def _deserialize_connection(self, conn_id, value):
```

Comment:
I think we should we make it `deserialize_connection` and explicity document it as overrideable by Secret Backend implementation. And add description on that to the Secret Backends. This is a great way to handle custom serialization methods if needed.

Reply:
sure.

Reply:
i made it "public"

i took a look and the docs for secrets backend itself are pretty sparse.  i'd like to leave it that way for now.  

i realize also that ultimately the "managing connections" page will need a revamp (because much of it assumes you are going to use URI).  but given that that's a larger effort, and nothing about this PR should interfere with any of that guidance (since it should be fully backward compatible), i'd like to defer that docs revamp for a followup if that sounds OK to you.